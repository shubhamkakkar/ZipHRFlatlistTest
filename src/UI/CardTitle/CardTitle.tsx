import React from 'react';
import { View, Text } from 'react-native';
import styles from './styleSheetCardTitle';
import theme from '../../theme';

type TCardTitle = {
  title: string;
  color?: string;
  center?: boolean;
};
export default function CardTitle({ title, center, color }: TCardTitle) {
  return (
    <View style={styles.containerCardTitle}>
      <View style={styles.cardTitleContainer}>
        <Text
          style={[
            styles.cardTitleText,
            { textAlign: center ? 'center' : 'left', color: color || theme.primaryColor },
          ]}
        >
          {title}
        </Text>
      </View>
    </View>
  );
}
