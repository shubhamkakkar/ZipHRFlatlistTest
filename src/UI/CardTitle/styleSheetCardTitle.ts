import { StyleSheet } from 'react-native';
import theme from '../../theme';

const styles = StyleSheet.create({
  containerCardTitle: {
    borderBottomWidth: 1,
    borderBottomColor: theme.primaryColor + 80,
    paddingBottom: 5,
    marginHorizontal: 10,
    paddingTop: 10,
  },
  cardTitleContainer: {
    flex: 1,
  },
  cardTitleText: {
    fontWeight: 'bold',
    fontSize: 16,
    lineHeight: 24,
  },
});

export default styles;
