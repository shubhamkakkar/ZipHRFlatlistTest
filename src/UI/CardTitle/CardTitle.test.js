import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import CardTitle from './CardTitle';

it('CardTitle matches snapshot - props: {{  title: "Title" }}', () => {
  const tree = renderer.create(<CardTitle {...{ title: 'Title' }} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('CardTitle matches snapshot - props: {{ title: "Title" , center: true }}', () => {
  const tree = renderer.create(<CardTitle {...{ title: 'Title', center: true }} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('CardTitle matches snapshot - props: {{ title: "Title", color: "red" }}', () => {
  const tree = renderer.create(<CardTitle {...{ title: 'Title', color: 'red' }} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('CardTitle matches snapshot - props: {{ title: "Title", color: "red"  , center: true}}', () => {
  const tree = renderer.create(<CardTitle {...{ title: 'Title', color: 'red', center: true }} />).toJSON();
  expect(tree).toMatchSnapshot();
});
