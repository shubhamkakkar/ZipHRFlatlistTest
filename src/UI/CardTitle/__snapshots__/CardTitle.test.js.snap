// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardTitle matches snapshot - props: {{  title: "Title" }} 1`] = `
<View
  style={
    Object {
      "borderBottomColor": "#7b2cbf80",
      "borderBottomWidth": 1,
      "marginHorizontal": 10,
      "paddingBottom": 5,
      "paddingTop": 10,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 16,
            "fontWeight": "bold",
            "lineHeight": 24,
          },
          Object {
            "color": "#7b2cbf",
            "textAlign": "left",
          },
        ]
      }
    >
      Title
    </Text>
  </View>
</View>
`;

exports[`CardTitle matches snapshot - props: {{ title: "Title" , center: true }} 1`] = `
<View
  style={
    Object {
      "borderBottomColor": "#7b2cbf80",
      "borderBottomWidth": 1,
      "marginHorizontal": 10,
      "paddingBottom": 5,
      "paddingTop": 10,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 16,
            "fontWeight": "bold",
            "lineHeight": 24,
          },
          Object {
            "color": "#7b2cbf",
            "textAlign": "center",
          },
        ]
      }
    >
      Title
    </Text>
  </View>
</View>
`;

exports[`CardTitle matches snapshot - props: {{ title: "Title", color: "red"  , center: true}} 1`] = `
<View
  style={
    Object {
      "borderBottomColor": "#7b2cbf80",
      "borderBottomWidth": 1,
      "marginHorizontal": 10,
      "paddingBottom": 5,
      "paddingTop": 10,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 16,
            "fontWeight": "bold",
            "lineHeight": 24,
          },
          Object {
            "color": "red",
            "textAlign": "center",
          },
        ]
      }
    >
      Title
    </Text>
  </View>
</View>
`;

exports[`CardTitle matches snapshot - props: {{ title: "Title", color: "red" }} 1`] = `
<View
  style={
    Object {
      "borderBottomColor": "#7b2cbf80",
      "borderBottomWidth": 1,
      "marginHorizontal": 10,
      "paddingBottom": 5,
      "paddingTop": 10,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 16,
            "fontWeight": "bold",
            "lineHeight": 24,
          },
          Object {
            "color": "red",
            "textAlign": "left",
          },
        ]
      }
    >
      Title
    </Text>
  </View>
</View>
`;
